src/transformStateWithClones.js

```diff
@@ -7,7 +7,53 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  actions.forEach((item, index, array) => {
+    let newObj;
+
+    switch (item.type) {
+      case 'addProperties':
+        newObj = result.length === 0
+          ? { ...state }
+          : { ...result[result.length - 1] };
+
+        for (const key in item.extraData) {
+          newObj[key] = item.extraData[key];
+        }
+        result.push(newObj);
+        break;
+
+      case 'removeProperties':
+        const newObjRemove = result.length === 0
+          ? { ...state }
+          : { ...result[result.length - 1] };
+
+        item.keysToRemove.forEach((key) => {
+          if (newObjRemove[key]) {
```

Comment:
Replace the check 'if (newObjRemove[key])' with 'if (key in newObjRemove)' to handle falsy property values.